client/src/components/Component.js

```diff
@@ -0,0 +1,27 @@
+export default class Component {
+  $dom;
+  state;
+  props;
+
+  constructor(props) {
+    this.props = props;
+  }
+
+  setState = (nextState) => {
+    this.state = nextState;
+    this.render();
+  };
+
+  createDom = (tagName, className) => {
+    this.$dom = document.createElement(tagName);
+    this.$dom.className = className;
+  };
```

Comment:
```suggestion
  /**
   * @param {Object} attrs
   */
  createDom = (tagName, attrs) => {
    this.$dom = document.createElement(tagName);
    for (const [key, value] of Object.entires(attrs)) {
      this.$dom[key] = value;
    }
  };
```

이런식으로 해서 여러개의 property를 설정할 수 있게 하면 좋겠습니다.

Reply:
좋은 생각인 것 같습니다! createDom 을 호출할 때도 속성 이름을 명시할 수 있어 좋은 것 같습니다.